modules/tensor_mechanics/src/dampers/ElementJacobianDamper.C

```diff
@@ -0,0 +1,136 @@
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "DisplacedProblem.h"
+#include "Assembly.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h" // _qrule
+
+template <>
+InputParameters
+validParams<ElementJacobianDamper>()
+{
+  InputParameters params = validParams<GeneralDamper>();
+  params.addClassDescription("Damper that limits the change in element Jacobians");
+  params.addParam<std::vector<NonlinearVariableName>>("displacements",
+                                                      "The nonlinear displacement variables");
+  params.addParam<Real>(
+      "max_increment",
+      0.1,
+      "The maximum permissible relative increment in the Jacobian per Newton iteration");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+ElementJacobianDamper::ElementJacobianDamper(const InputParameters & parameters)
+  : GeneralDamper(parameters),
+    _tid(parameters.get<THREAD_ID>("_tid")),
+    _assembly(_subproblem.assembly(_tid)),
+    _qrule(_assembly.qRule()),
+    _JxW(_assembly.JxW()),
+    _fe_problem(*parameters.getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")),
+    _displaced_problem(_fe_problem.getDisplacedProblem()),
+    _max_jacobian_diff(parameters.get<Real>("max_increment"))
+{
+  if (_displaced_problem == NULL)
+    mooseError("ElementJacobianDamper: Must use displaced problem");
+
+  _mesh = &_displaced_problem->mesh();
+
+  const std::vector<NonlinearVariableName> & nl_vnames(
+      getParam<std::vector<NonlinearVariableName>>("displacements"));
+  _ndisp = nl_vnames.size();
+
+  for (unsigned int i = 0; i < _ndisp; ++i)
+  {
+    _disp_var.push_back(&_sys.getVariable(_tid, nl_vnames[i]));
+    _disp_incr.push_back(_disp_var.back()->increment());
+  }
+}
+
+void
+ElementJacobianDamper::initialSetup()
+{
+}
+
+Real
+ElementJacobianDamper::computeDamping(const NumericVector<Number> & /* solution */,
+                                      const NumericVector<Number> & update)
+{
+  // Maximum difference in the Jacobian for this Newton iteration
+  Real max_difference = 0.0;
+  MooseArray<Real> JxW_displaced;
+
+  // Vector for storing node coordinates before displacing them
+  std::vector<Point> node_copies;
+
+  // Loop over elements in the mesh
+  const MeshBase::element_iterator end = _mesh->getMesh().active_local_elements_end();
+  for (auto el = _mesh->getMesh().active_local_elements_begin(); el != end; ++el)
+  {
+    Elem * current_elem = *el;
+
+    node_copies.clear();
+    node_copies.reserve(current_elem->n_nodes());
+
+    // Displace nodes with current Newton increment
+    for (unsigned int i = 0; i < current_elem->n_nodes(); ++i)
+    {
+      Node & displaced_node = current_elem->node_ref(i);
+
+      node_copies.push_back(displaced_node);
+
+      for (unsigned int j = 0; j < _ndisp; ++j)
+      {
+        dof_id_type disp_dof_num =
+            displaced_node.dof_number(_sys.number(), _disp_var[j]->number(), 0);
+        displaced_node(j) += update(disp_dof_num);
```

Comment:
Ah, I see. @bwspenc and I had chatted yesterday and I was under the impression that we were going to clone the element and nodes and displace them so that we could leave the mesh untouched. The reason for this was to make it thread-safe. That might be phase 2 though. I see that it's using a GeneralDamper so there aren't any issues with it as-is. It's just less scalable for now.